src/components/LiquidityTable.tsx

```diff
@@ -5,54 +5,74 @@ import { getCoinGeckoApi } from 'utils/constants/constants'
 import { ExchangeName } from 'utils/poolData'
 import ExchangeSummary from './ExchangeSummary'
 import { TokenContext } from 'contexts/Token'
-import { PRICE_DECIMALS, EXCHANGES } from 'utils/constants/constants'
+import {
+  ChainId,
+  COIN_GECKO_CHAIN_KEY,
+  PRICE_DECIMALS,
+  EXCHANGES,
+} from 'utils/constants/constants'
 
-const LiquidityTable = (props: {
-  desiredAmount: string
-  networkKey: string
-}) => {
+const LiquidityTable = (props: { chainId: ChainId; desiredAmount: string }) => {
   const [tokenPrice, setTokenPrice] = useState<BigNumber>(BigNumber.from(0))
   const { selectedToken } = useContext(TokenContext)
   const [tokenPriceLoading, setTokenPriceLoading] = useState(true)
-  const [prevTokenAddress, setSelectedTokenAddress] = useState("")
+  const [prevTokenAddress, setSelectedTokenAddress] = useState('')
 
-  const didTokenUpdate = (prevTokenAddresss: string, presTokenAddress: string) => {
+  const didTokenUpdate = (
+    prevTokenAddresss: string,
+    presTokenAddress: string
+  ) => {
     if (prevTokenAddresss !== presTokenAddress) {
       setTokenPriceLoading(true)
       setSelectedTokenAddress(presTokenAddress)
     }
   }
-  didTokenUpdate(prevTokenAddress,selectedToken.address)
+  didTokenUpdate(prevTokenAddress, selectedToken.address)
 
   // get token price in USD
   useEffect(() => {
-    fetch(getCoinGeckoApi(selectedToken.address, props.networkKey))
+    const networkKey = COIN_GECKO_CHAIN_KEY[props.chainId]
+    fetch(getCoinGeckoApi(selectedToken.address, networkKey))
       .then((response) => response.json())
       .then((response) => {
+        console.log(response)
+        console.log(
+          '-->',
+          selectedToken.address,
+          response[selectedToken.address.toLowerCase()]?.usd,
+          networkKey
+        )
         setTokenPrice(
           BigNumber.from(
-            Math.round(response[selectedToken.address]?.usd * PRICE_DECIMALS)
+            Math.round(
+              response[selectedToken.address.toLowerCase()]?.usd *
```

Comment:
Found this needs to be lower case as otherwise the address sometimes might not be recognized in the response.